File: frontend/src/services/authService.js
import api from "./api";

export const register = (data) => api.post("/auth/register", data);
export const login = (data) => api.post("/auth/login", data);
export const logout = (refreshToken) => api.post(`/auth/logout?refreshToken=${refreshToken}`);
export const refreshToken = (token) => api.post(`/auth/refresh?refreshToken=${token}`);

File: src/main/java/com/example/Techwing/service/AuthService.java
package com.example.Techwing.service;

import com.example.Techwing.models.User;
import com.example.Techwing.payload.AuthResponse;
import com.example.Techwing.payload.LoginRequest;
import com.example.Techwing.payload.RegisterRequest;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
    AuthResponse refreshToken(String refreshToken);
    void logout(String refreshToken);
    User getCurrentUser(String email);
}
